cpp/CMakeLists.txt

```diff
@@ -503,11 +505,10 @@ if(BUILD_CUML_CPP_LIBRARY)
   )
 
   target_link_libraries(${CUML_CPP_TARGET}
-    PUBLIC
+    PRIVATE
```

Comment:
I'm not sure I follow this change, since the raft handle is in our public API, shouldn't `raft::raft` be a `PUBLIC` dependency?